src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -523,6 +523,14 @@ theorem getElem?_zeroExtend (m : Nat) (x : BitVec n) (i : Nat) :
   all_goals (first | apply getLsbD_ge | apply Eq.symm; apply getLsbD_ge)
     <;> omega
 
+theorem getElem_truncate (m : Nat) (x : BitVec n) (i : Nat) (hi : i < m) :
+    (truncate m x)[i] = x.getLsbD i := by
+  simp [getElem?_eq, getElem_zeroExtend]
```

Comment:
This one can just be `x[i]` on the right hand side.

My suggestion (not entirely sure I've followed it previously, but I think) is to prefer `x[i]` over `.getLsbD` on the RHS where possible --- but when it isn't possible without an `if` or `getD`, accepting `getLsbD`.

Reply:
AFAIU, this can only be `x[i]` on the RHS if I introduce `h : i < n`. For some reason, truncate is just an alias to `zeroExtend`, so there is no knowledge that `i < x`. Or maybe I miss something here?

I implemented this change by adding the `h : i < n` hypothesis. This does not feel that beautiful, though.

Reply:
Oh, sorry, I misread the theorem. Sorry, no change required here.